### ISSUE TITLE
Feature request: Reading Aseprite files and creating texture atlas(es) directly in KorGE 

### ISSUE THREAD

**Issue by jobe-m** [korlibs/korge-next#314]
This is for gathering ideas and requirements for implementing import of Aseprite (ase) files into KorGE and creating texture atlases out of them.

- [ ] Minimize the generated amount of texture atlases to a minimum so that batch rendering can happen as performant as possible. Thus pack multiple ase files into one texture atlas. Create new texture atlas when the first atlas does not have any space left.
- [ ] Possibility to split tag frames of one ase file into different sprites in the generated texture atlas.
- [ ] Possibility to split layers of one ase file into different sprites in the generated texture atlas.
- [ ] Possibility to split slices within one or more specific layer(s) and/or tag(s) into different sprites in the generated texture atlas.
- [ ] Possibility to ignore empty (tag) frames or layers. Thus do not generate a sprite in the texture atlas.
- [ ] Merge duplicate sprite images in the generated texture atlas to save space.
- [ ] Specify padding/spacing in pixel between sprites in the texture atlas and between sprites and the texture border.
- [ ] Specify per ase file if the generated sprites are extruded in the generated texture atlas.
- [ ] Trim sprite images (remove empty space around sprites) to save space in the generated texture atlas. Make sure to specify the correct offset for those sprites in the JSON file.
- [ ] Read tileset(s) from ase file (v1.3) and place the tiles inside generated texture atlas. The tileset should be accessible by the name of the layer where it is used. (Aseprite can have different tilesets per layer in one ase file.)
- [ ] Possibility to extrude the tiles within a tileset. (This is useful to prevent seams when scrolling the tiled view.)
- [ ] Provide a build plugin to generate texture atlases at build time instead of at run time.

**Comment by jobe-m** [comment]
Added:
- Do not include hidden layers of ASE file into generated atlas
This is particularly useful since I often use additional layers for drawing helper lines or similar. That should of course not go into the game.

**Comment by jobe-m** [comment]
Added:
- Do not draw hidden layers of ASE file

With the last PR I could remove hidden layers from being stored in the atlas texture. But for some reason the hidden layers are still drawn on the screen with the "ImageDataView" object. For me it looks like that the hidden layers are drawn from the original texture object? So we need to check how we can prevent drawing of those hidden layers. In the end hidden layers should just be ignored completely in KorIM ASE.kt implementation.

**Comment by soywiz** [comment]
Yeah, thinking about that, the `includeInAtlas` is just to copy that image into the atlas, not for disabling it.
I added it to not include the whole canvas into the atlas. But the images are still there, but in a separate Bitmap.

Probably it will be better to just filter/remove the layers or propagate the visibility information to the image data, so we can filter it when rendering. By either setting view.visible to false, or by just not creating or setting the Image object.

**Comment by jobe-m** [comment]
I guess for a first version it would be good to ignore / remove the layer when it is hidden. Maybe that could be configurable with the readImageDatacontainer call IIRC the function name. Can you point me to the code which creates the Image object from the layer data? I would look into it. I guess it's in ASE.kt?

**Comment by soywiz** [comment]
It looks like there is some duplicated code :S, one for the root and other for the slices.

(You might need to do this twice):

* You can probably `.filter { }` here: https://github.com/korlibs/korge-next/blob/bbbceb857c7fa4107eedf72f224a3f02e8ceb08e/korim/src/commonMain/kotlin/com/soywiz/korim/format/ASE.kt#L38
* And here: https://github.com/korlibs/korge-next/blob/bbbceb857c7fa4107eedf72f224a3f02e8ceb08e/korim/src/commonMain/kotlin/com/soywiz/korim/format/ASE.kt#L39
* By this property: https://github.com/korlibs/korge-next/blob/bbbceb857c7fa4107eedf72f224a3f02e8ceb08e/korim/src/commonMain/kotlin/com/soywiz/korim/format/ASE.kt#L54

Alternatively:

* Add a visible property here: https://github.com/korlibs/korge-next/blob/bbbceb857c7fa4107eedf72f224a3f02e8ceb08e/korim/src/commonMain/kotlin/com/soywiz/korim/format/ImageLayer.kt
* Set the visible property here: https://github.com/korlibs/korge-next/blob/bbbceb857c7fa4107eedf72f224a3f02e8ceb08e/korge/src/commonMain/kotlin/com/soywiz/korge/view/animation/ImageAnimationView.kt#L45-L48

When visible=false, the image doesn't enter in the batch, so even if it is in a separate bitmap, no draw or batch computation will happen.
